### PR TITLE
Add some RAKwireless WisBlock based devices

### PR DESCRIPTION
**Pull request by beegee-tokyo**
@brocaar 

I added some basic RAKwireless devices. Please check if we are doing it right.
About the "profiles", these devices are using LoRaMAC stack V1.0.2. I chose a naming that allows to define more profiles for LoRaMAC stack V1.0.3 and V1.0.4 that we will add later.

### CODE REVIEW

File: vendors/rakwireless/codecs/test_decode_example.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[
```
Comment (brocaar) [inline vendors/rakwireless/codecs/test_decode_example.json]: This file must be named: `test_decode_rak_standard.json` (as it is related to the `rak_standard.js` codec file).

File: vendors/rakwireless/codecs/test_encode_example.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[
```
Comment (brocaar) [inline vendors/rakwireless/codecs/test_encode_example.json]: This file must be named: `test_encode_rak_standard.json` (as it is related to the `rak_standard.js` codec file).
If there are no tests, you can leave the content to `[]` (empty array of tests).

File: vendors/rakwireless/devices/wisblock-api-v2.toml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  # Device firmware version.
+  #
+  # This section can be repeated in case multiple firmware versions exist.
+  # As a new firmware version can change the supported profiles / regions and
+  # payload format, each firmware version has its own profiles and codec
+  # configuration.
+  [[device.firmware]]
+    # Firmware version.
+    version = "2.0.18^"
```
Comment (brocaar) [inline vendors/rakwireless/devices/wisblock-api-v2.toml]: Is the `^` character intentionally? See also the other devices.

**Review by brocaar — CHANGES_REQUESTED** [review]
Thanks :+1: I have added a few comments.

File: vendors/rakwireless/codecs/test_decode_example.json (hunk 1 above)
Comment (beegee-tokyo) [reply vendors/rakwireless/codecs/test_decode_example.json]: OK, got it.

File: vendors/rakwireless/codecs/test_encode_example.json (hunk 2 above)
Comment (beegee-tokyo) [reply vendors/rakwireless/codecs/test_encode_example.json]: OK, got it.

File: vendors/rakwireless/devices/wisblock-api-v2.toml (hunk 3 above)
Comment (beegee-tokyo) [reply vendors/rakwireless/devices/wisblock-api-v2.toml]: Yes, the `^` is intentional. I got it from PlatformIO, where it means "this release or newer if backward compatible".
I can change it and remove the `^` (I used it the wrong way anyway).

**Comment by beegee-tokyo** [comment]
@brocaar 
Changed the file names and removed the `^`

**Comment by brocaar** [comment]
@beegee-tokyo the test is failing:

> Sensor type error!: 66

Could you please check (make sure that the input bytes are integers, not hex).

**Comment by beegee-tokyo** [comment]
@brocaar 
Can you try again

**Comment by brocaar** [comment]
The codec tests is passing now, but it is failing on `AS923_1` because in the validation I only added `AS923-1`, which is the official common-name, but ChirpStack itself expects `AS923_1` (I think that was because of Go / Rust `const` to string mapping). I will update the validation to for now accept both values and then update ChirpStack to use the official naming.

**Comment by brocaar** [comment]
(I deleted my prev. comment).


Could you update the common-names to match the Regional Parameters values? Sorry I have missed this in my review:

image


(`AS923_1` -> `AS923`, `AS923_2` -> `AS923-2`, ...)

**Comment by beegee-tokyo** [comment]
@brocaar 
Done.

**Review by brocaar — APPROVED** [review]

**Comment by brocaar** [comment]
Thanks @beegee-tokyo for adding the first real device :-) All tests are passing and I have merged your PR.

**Comment by beegee-tokyo** [comment]
🥳

**Pull request merged**